setup.py

```diff
@@ -9,35 +9,27 @@
 from setuptools import find_packages, setup
 
 
-# Dependencies of GPflow
-requirements = [
-    "numpy>=1.10.0",
-    "scipy>=0.18.0,==1.4.1",  # pinned to ==1.4.1 to satisfy tensorflow requirements
-    "multipledispatch>=0.6",
-    "tabulate",
-    "typing_extensions",
-]
-
-if sys.version_info < (3, 7):
-    # became part of stdlib in python 3.7
-    requirements.append("dataclasses")
+##### Dependencies of GPflow
 
 # We do not want to install tensorflow in the readthedocs environment, where we
 # use autodoc_mock_imports instead. Hence we use this flag to decide whether or
 # not to append tensorflow and tensorflow_probability to the requirements:
 if os.environ.get("READTHEDOCS") != "True":
-    requirements.extend(
-        [
-            # tensorflow>=2.3 not compatible with tensorflow-probability<0.11
-            "tensorflow>=2.1.0,<2.3",
-            # tensorflow-probability==0.10.0 doesn't install correctly
-            # https://github.com/tensorflow/probability/issues/991
-            #
-            # gpflow uses private functionality not present in tensorflow-probability~=0.11
-            "tensorflow-probability>0.10.0,<0.11",
-            "setuptools>=41.0.0",  # to satisfy dependency constraints
-        ]
-    )
+    requirements = [
+        "tensorflow>=2.1.0",
+        "tensorflow-probability>0.10.0",  # tensorflow-probability==0.10.0 doesn't install correctly, https://github.com/tensorflow/probability/issues/991
+        "setuptools>=41.0.0",  # to satisfy dependency constraints
+    ]
+
+else:
+    requirements = []
+
+requirements.extend(
+    ["numpy", "scipy", "multipledispatch>=0.6", "tabulate", "typing_extensions",]
+)
```

Comment:
i'd be surprised if this isn't enough to get it to build, so you could try it without the 2020-resolver

Reply:
i think the problem is that pip currently doesn't update a dependency if it's further constrained later, so
```
numpy>1.10.0
numpy<1.11.0
```
might happily install 1.12

Reply:
ok, turns out we do need the new resolver :(

Reply:
that's odd. I don't see this problem locally